storm-core/src/jvm/backtype/storm/grouping/PartialKeyGrouping.java

```diff
@@ -65,7 +66,11 @@ public void prepare(WorkerTopologyContext context, GlobalStreamId stream, List<I
                 List<Object> selectedFields = outFields.select(fields, values);
                 ByteBuffer out = ByteBuffer.allocate(selectedFields.size() * 4);
                 for (Object o: selectedFields) {
-                    out.putInt(o.hashCode());
+                    if (o instanceof Object[]) {
```

Comment:
@d2r 
Could we add same thing (converting List to Object[] by toArray, and apply Arrays.deepHashCode()) from list-hash-code to this?